kotlin/codegen/src/main/java/com/squareup/moshi/kotlin/codegen/AdapterGenerator.kt

```diff
@@ -133,9 +133,16 @@ internal class AdapterGenerator(
     }
 
     result.addProperty(optionsProperty)
+
+    val delegateKeyCount = propertyList.groupingBy { it.delegateKey }.eachCount()
+
     for (uniqueAdapter in propertyList.distinctBy { it.delegateKey }) {
+      // If the adapter is shared for multiple properties,
+      // don't provide a the fieldName hint to the adapter lookup.
```

Comment:
looking at this again, i realize this is wrong. it might as well have a hint.
if you have
```
@JsonClass(generateAdapter = true)
class HasPlatformType(val strings: ArrayList<String>, val list: ArrayList<String>)
```
the error message might as well tell you that you have an error in `strings` instead of not saying anything.